UI/src/app/components/popups/open-cmd-filter/cmd-filter.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+import  *  as  data  from  './CdmByTypes.json';
+import { MatListOption } from '@angular/material/list';
+import { uniq } from 'src/app/infrastructure/utility';
+
+
+
+@Component({
+  selector: 'app-cmd-filter',
+  templateUrl: './cmd-filter.component.html',
+  styleUrls: ['./cmd-filter.component.scss']
+})
+export class CmdFilterComponent implements OnInit {
+
+  targetTypes: string [] = []
+  cdmByTypes:Map<string, string[]>
+  selectedTables:string[] = []
+  selectedTypes:string[] = []
+  @Output() complete = new EventEmitter<void>();
+  
+  constructor() { }
+
+  ngOnInit() {
+    this.cdmByTypes = (data as any).default
+    this.targetTypes.push("Show All")
+    this.targetTypes = this.targetTypes.concat(uniq(Object.keys(this.cdmByTypes)))
+  }
+
+  onTypeSelection(types: MatListOption[]){
+    this.selectedTables=[]
+    this.selectedTypes = types.map(item => item.value)
```

Comment:
What was the point of this action here? You already reset this.selectedTypes inside of for every time